File: AdventOfCode/2025/Day4/Python version/Part1.py
try:
    with open("input.txt", "r") as f:
        lines = []
        for line in f:
            stripped = line.rstrip('\n')
            if stripped != "":
                lines.append(stripped)
except FileNotFoundError:
    print("ERROR")
    exit()

if not lines:
    print(0)
    exit()



grid = [list(line) for line in lines]
rows = len(grid)
column = len(grid[0])
accessible = 0
directions = [(-1, -1), (-1, 0), (-1, 1),(0, -1), (0, 1), (1, -1),  (1, 0),  (1, 1)]


for i in range(rows):
    for j in range(column):
        if grid[i][j] == '@':
            count = 0
            for dx, dy in directions:
                ni = i + dx
                nj = j + dy
                if 0 <= ni < rows and 0 <= nj < column:
                    if grid[ni][nj] == '@':
                        count += 1
            if count < 4:
                accessible += 1

print(accessible)

File: AdventOfCode/2025/Day4/Part1.cpp
#include <bits/stdc++.h>
#include <fstream>
using namespace std;

int main() {
    ios_base::sync_with_stdio(false);
    cin.tie(nullptr);

    ifstream file("input.txt");
    if (!file.is_open()) return cerr << "ERROR", 1;

    vector<string> grid;
    string line;

    while (getline(file, line)) {
        if (line.empty()) continue;
        grid.push_back(line);
    }

    if (grid.empty()) {
        cout << 0 << '\n';
        return 0;
    }

    int rows = grid.size();
    int column = grid[0].size();
    int accessible = 0;

    vector<int> dx = {-1, -1, -1, 0, 0, 1, 1, 1};
    vector<int> dy = {-1, 0, 1, -1, 1, -1, 0, 1};

    for (int i = 0; i < rows; ++i) {
        for (int j = 0; j < column; ++j) {
            if (grid[i][j] != '@') continue;

            int count = 0;
            for (int k = 0; k < 8; ++k) {
                int nx = i + dx[k];
                int ny = j + dy[k];
                if (nx >= 0 && nx < rows && ny >= 0 && ny < column && grid[nx][ny] == '@') {
                    count++;
                }
            }

            if (count < 4) {
                accessible++;
            }
        }
    }

    cout << accessible << '\n';
    return 0;
}
